management/vmutils/rolestate.go

```diff
@@ -0,0 +1,31 @@
+package vmutils
+
+import (
+	"time"
+
+	"github.com/MSOpenTech/azure-sdk-for-go/management"
+	vm "github.com/MSOpenTech/azure-sdk-for-go/management/virtualmachine"
+)
+
+// Blocks until all role instances in deployment to reach desired power state
+func WaitForDeploymentPowerState(client management.Client, cloudServiceName, deploymentName string, desiredPowerstate vm.PowerState) error {

```

Comment:
I need PowerState for my purpose. I can create another PR for DeploymentStatus and role InstanceStatus (there is nothing called RoleState in the SDK?)